csharp-ovh/Client/Client.cs

```diff
@@ -347,60 +377,40 @@ private string Call(string method, string path, string data = null, bool needAut
         /// <param name="isBatch">If true, this will query multiple resources in one call</param>
         /// <exception cref="HttpException">When underlying request failed for network reason</exception>
         /// <exception cref="InvalidResponseException">when API response could not be decoded</exception>
-        private Task<string> CallAsync(string method, string path, string data = null, bool needAuth = true, bool isBatch = false)
+        private async Task<string> CallAsync(string method, string path, string data = null, bool needAuth = true, bool isBatch = false)
         {
-            PrepareCall(ref method, ref path, data, needAuth, isBatch);
-
-            Task<string> responseTask = null;
-
-            if (method != "GET")
-            {
-                responseTask = _webClient.UploadStringTaskAsync(path, method, data ?? "");
-            }
-            else
+            if (path.StartsWith("/"))
             {
-                responseTask = _webClient.DownloadStringTaskAsync(path);
+                path = path.Substring(1);
             }
 
-            return responseTask.ContinueWith((r) => CallAsyncError(r));
+            HttpResponseMessage response = null;
 
-            // Wrapper used so that we catch the exception ourselves
-            // and return something more useful to the user if we can
-            string CallAsyncError(Task<string> task)
+            try
             {
-                try
+                var httpMethod = new HttpMethod(method);
+                HttpRequestMessage msg = new HttpRequestMessage (httpMethod, new Uri(Endpoint + path));
+                if(httpMethod != HttpMethod.Get)
                 {
-                    return task.Result;
-                }
-                catch (WebException ex)
-                {
-                    throw HandleWebException(ex);
+                    if(data != null)
+                    {
+                        msg.Content = new StringContent(data ?? "", Encoding.UTF8, "application/json");
+                    }
                 }
+                SetHeaders(msg, method, Endpoint + path, data, needAuth, isBatch);
+                response = await _httpClient.SendAsync(msg).ConfigureAwait(false);
             }
-        }
-
-        private Exception HandleWebException(WebException ex)
-        {
-            using (HttpWebResponse httpResponse = (HttpWebResponse)ex.Response)
+            catch(HttpRequestException e)
             {
-                if (httpResponse == null)
-                {
-                    return new HttpException("Low HTTP request failed error", ex);
-                }
-                return ExtractExceptionFromHttpError(httpResponse, ex);
+                throw new HttpException("An exception occured while trying to issue the HTTP call", e);
```

Comment:
Why not directly throwing the `HttpRequestException`?

Reply:
Previously, to stick with the exception names and behaviour of python-ovh.
Now, because it would be a breaking change. I might do it for 4.0 though...